lib/db.js

```diff
@@ -22,10 +23,91 @@ class DB {
                 statement.finalize();
             }
         });
+        /* Process the array of storage groups declared in the config */
+        this._storageGroups = this._buildStorageGroups(this.conf.storage_groups);
+        /* The cache holding the already-resolved domain-to-group mappings */
+        this._storageGroupsCache = new Map();
+    }
+
+    /**
+     * Process the storage group configuration.
+     * @param {Array} groups the array of group objects to read, each must contain
+     *                at least the name and domains keys
+     * @return {Array} Array of storage group objects
+     */
+    _buildStorageGroups(groups) {
+        const storageGroups = [];
+        if (!Array.isArray(groups)) {
+            return storageGroups;
+        }
+        groups.forEach((group) => {
+            const grp = extend(true, {}, group);
+            if (!Array.isArray(grp.domains)) {
+                grp.domains = [grp.domains];
+            }
+            grp.domains = grp.domains.map((domain) => {
+                if (/^\/.*\/$/.test(domain)) {
+                    return new RegExp(domain.slice(1, -1));
+                }
+                return domain;
+            });
+            storageGroups.push(grp);
+        });
+        return storageGroups;
+    }
+
+    /**
+     * Finds the storage group for a given domain.
+     * @param  {string} domain  the domain's name
+     * @return {Object}         the group object matching the domain
+     */
+    _resolveStorageGroup(domain) {
+        if (this._storageGroupsCache.has(domain)) {
+            return this._storageGroupsCache.get(domain);
+        }
+        let idx;
+        let group;
+
+        // not found in cache, find it
+        for (idx = 0; idx < this._storageGroups.length; idx++) {
+            const curr = this._storageGroups[idx];
+            let domIdx;
+            for (domIdx = 0; domIdx < curr.domains.length; domIdx++) {
+                const dom = curr.domains[domIdx];
+                if (((dom instanceof RegExp) && dom.test(domain)) ||
+                    (typeof dom === 'string' && dom === domain)) {
+                    group = curr;
+                    break;
+                }
+            }
+            if (group) {
+                break;
+            }
+        }
+        if (!group) {
+            // no group found, assume the domain is to
+            // be grouped by itthis
```

Comment:
itthis? This looks like a `s/self/this/` gone wrong

Reply:
'find' and 'replace all' going wrong. will fix.

Reply:
Still not fixed it seems